src/App.js

```diff
@@ -1,25 +1,59 @@
-import logo from './logo.svg';
+import { useState } from 'react';
 import './App.css';
 
+import foods from './foods.json';
+
+import FoodBox from './components/FoodBox';
+import AddNewFood from './components/AddNewFood';
+import Searchbar from './components/Searchbar';
+
 function App() {
+  const [food, setFood] = useState(foods);
+  const [showFood, setShowFood] = useState(foods);
+  /* console.log(foods) */
+  const addToList = (foods) => {
+    const foodCopy = [foods, ...food];
+    setShowFood(foodCopy);
+  };
+
+  const searchFoods = (query) => {
+    const filteredFoods = foods.filter((food) =>
+      food.name.toLowerCase().includes(query.toLowerCase())
+    );
+    setShowFood(filteredFoods);
+  };
+
+  const deleteFood = (name) => {
+    const filteredFoods = foods.filter((food) => {
+      return food.name !== name;
+    });
+    setShowFood(filteredFoods);
```

Comment:
Your delete food is not working properly because:
- you're filtering through `foods` (json file) instead of `food` (variable);
- you're only updating `showFood` with `setShowFood` you need to update `food` as well;
```
  const deleteFood = (name) => {
    const filteredFoods = food.filter((food) => {
      return food.name !== name });
    setShowFood(filteredFoods);
    setFood(filteredFoods);
  };
```